composition-js/src/validate.ts

```diff
@@ -474,7 +479,16 @@ export class ValidationState {
       return { error: satisfiabilityError(newPath, this.subgraphPaths.map((p) => p.path), deadEnds) };
     }
 
-    const updatedState = new ValidationState(newPath, newSubgraphPaths);
+    // If the edge has an override condition, we should capture it in the state so
+    // that we can ignore later edges that don't satisfy the condition.
+    const newOverrideCondition = supergraphEdge.overrideCondition ? {
```

Comment:
Earlier in this function, there's a call to `advancePathWithTransition()`. That code has to take in this `newOverrideCondition`, so that it can take the corresponding virtual edge in the subgraph, so you'll need to move this `newOverrideCondition` definition to before that `advancePathWithTransition()` call.

Within `advancePathWithTransition()`, you'll pass `newOverrideCondition` to `advancePathWithDirectTransition()`. Then in `advancePathWithDirectTransition()`, specifically [here](https://github.com/apollographql/federation/blob/c83c25a430327da5caa0b04db79b575420e88e50/query-graphs-js/src/graphPath.ts#L1577), you'll need to:
1. Check if the subgraph edge has an override condition.
2. If it does, assert that the `newOverrideCondition` has a value for the corresponding label (this is more just sanity checking).
3. If the `XOR` of the edge's override condition boolean and the `newOverrideCondition`'s boolean is true, then `continue`/skip the rest of that `for` loop.

Reply:
I may have misunderstood, or maybe it's fine as is, you tell me. I passed all conditions rather than just the new one.
[dbb26c1](https://github.com/apollographql/federation/pull/2902/commits/dbb26c1b11cf6c3f10490c83121a0df9504078c4)

Reply:
Introducing this started failing my fpf test with some errors. I experimented a bit with that `continue`, by doing nothing, returning an unadvanceable, and finally trying out a different "bypass" approach, all of which cause different errors that I'm unsure about. Maybe we can take a look tomorrow.

This commit is interesting.
https://github.com/apollographql/federation/pull/2902/commits/d1a5bf719e5e170cbe5e2532327f02620f7950c0

Reply:
Resolved by Sachin in [674bb17](https://github.com/apollographql/federation/pull/2902/commits/674bb17c4487ea78ae8230c7321e0f0f40f66bfd)